### PR TITLE
operator: add operator_authorized_github_accounts parameter

### PR DESCRIPTION
**Pull request by berendt**
Supports a list of Github accounts from which the public keys are added to the authorized keys.

Related to SovereignCloudStack/issues#433

### CODE REVIEW

**Review by artificial-intelligence — APPROVED** [review]
LGTM

I don't know if there are limits on the kind of keys that are supported (e.g. only rsa), if that would be the case, that should be documented, but can be done later - if there even are limitiations.

**Comment by berendt** [comment]
> LGTM
> 
> I don't know if there are limits on the kind of keys that are supported (e.g. only rsa), if that would be the case, that should be documented, but can be done later - if there even are limitiations.

I am not aware of any limitations at the moment.

File: roles/operator/tasks/main.yml [hunk 1]
```diff
@@ -65,6 +65,14 @@
   loop: "{{ operator_authorized_keys }}"
   no_log: true
 
+- name: Set authorized github accounts
+  become: true
+  ansible.posix.authorized_key:
+    key: "{{ lookup('url', 'https://github.com/' + item + '.keys', split_lines=False) }}"
+    user: "{{ operator_user }}"
+  loop: "{{ operator_authorized_github_accounts }}"
+  no_log: true
```
Comment (osfrickler) [inline roles/operator/tasks/main.yml:74]: The keys are pulled from a public location, using `no_log` here seems wrong.

File: roles/operator/README.rst [hunk 2]
```diff
@@ -32,6 +32,11 @@ The default shell for the operator.
 
 A list of ssh authorized keys to add.
```
Comment (osfrickler) [inline roles/operator/README.rst:33]: Maybe reword this while at it?

`List of SSH public keys to add to the authorized keys for the operator account.`

File: roles/operator/README.rst [hunk 3]
```diff
@@ -32,6 +32,11 @@ The default shell for the operator.
...
 
+.. zuul:rolevar:: operator_authorized_github_accounts
+   :default: []
+
+List of Github accounts from which the public keys are added to the authorized keys.
```
Comment (osfrickler) [inline roles/operator/README.rst:38]: "... the SSH public keys ..."? just to avoid confusion with GPG keys

File: roles/operator/tasks/main.yml [hunk 4]
```diff
@@ -65,6 +65,14 @@
   loop: "{{ operator_authorized_keys }}"
   no_log: true
```
Comment (osfrickler) [inline roles/operator/tasks/main.yml:66]: One could question this `no_log`, too

Comment (berendt) [reply roles/operator/tasks/main.yml:66]: Will change this with a second PR.

**Review by osfrickler — APPROVED** [review]
